diff --git a/bin/RegisterTritonDialects.h b/bin/RegisterTritonDialects.h
index 5ca0f2168..75860a618 100644
--- a/bin/RegisterTritonDialects.h
+++ b/bin/RegisterTritonDialects.h
@@ -39,6 +39,9 @@
 #include "mlir/Dialect/LLVMIR/Transforms/InlinerInterfaceImpl.h"
 #include "mlir/InitAllPasses.h"
 
+#include "cpu/include/Dialect/TritonCPU/IR/Dialect.h"
+#include "cpu/include/TritonCPUToLLVM/Passes.h"
+
 #include "mlir/Conversion/ArithToLLVM/ArithToLLVM.h"
 #include "mlir/Conversion/ControlFlowToLLVM/ControlFlowToLLVM.h"
 #include "mlir/Conversion/MathToLLVM/MathToLLVM.h"
@@ -135,6 +138,11 @@ inline void registerTritonDialects(mlir::DialectRegistry &registry) {
   mlir::triton::proton::gpu::registerScheduleBufferStorePass();
   mlir::triton::proton::gpu::registerAddSchedBarriersPass();
 
+  // CPU
+  mlir::triton::cpu::registerConvertTritonCPUToLLVM();
+  mlir::triton::cpu::registerAllocateSharedMemoryCPU();
+  mlir::triton::cpu::registerConvertMaskedOpsToLLVM();
+
   registry.insert<
       mlir::triton::TritonDialect, mlir::cf::ControlFlowDialect,
       mlir::triton::nvidia_gpu::TritonNvidiaGPUDialect,
@@ -146,5 +154,5 @@ inline void registerTritonDialects(mlir::DialectRegistry &registry) {
       mlir::triton::amdgpu::TritonAMDGPUDialect,
       mlir::triton::proton::ProtonDialect,
       mlir::triton::proton::gpu::ProtonGPUDialect, mlir::ROCDL::ROCDLDialect,
-      mlir::triton::gluon::GluonDialect>();
+      mlir::triton::gluon::GluonDialect, mlir::triton::cpu::TritonCPUDialect>();
 }
