diff --git a/bin/RegisterTritonDialects.h b/bin/RegisterTritonDialects.h
index 48049f026..c53213c93 100644
--- a/bin/RegisterTritonDialects.h
+++ b/bin/RegisterTritonDialects.h
@@ -39,6 +39,16 @@
 #include "mlir/Dialect/LLVMIR/Transforms/InlinerInterfaceImpl.h"
 #include "mlir/InitAllPasses.h"
 
+#include "cpu/include/Dialect/TritonCPU/IR/Dialect.h"
+#include "cpu/include/Dialect/TritonCPU/Transforms/Passes.h"
+#include "cpu/include/TritonCPUToLLVM/Passes.h"
+#include "npu/include/Dialect/TritonTenstorrent/IR/Dialect.h"
+#include "npu/include/Dialect/TritonTenstorrent/Transforms/Passes.h"
+#include "npu/include/TritonNPUToTenstorrent/Passes.h"
+
+#include "third_party/tt-mlir/include/ttmlir/Dialect/TTKernel/IR/TTKernel.h"
+#include "third_party/tt-mlir/include/ttmlir/Dialect/TTCore/IR/TTCore.h"
+
 #include "mlir/Conversion/ArithToLLVM/ArithToLLVM.h"
 #include "mlir/Conversion/ControlFlowToLLVM/ControlFlowToLLVM.h"
 #include "mlir/Conversion/MathToLLVM/MathToLLVM.h"
@@ -136,6 +146,23 @@ inline void registerTritonDialects(mlir::DialectRegistry &registry) {
   mlir::triton::proton::gpu::registerScheduleBufferStorePass();
   mlir::triton::proton::gpu::registerAddSchedBarriersPass();
 
+  // CPU
+  mlir::triton::cpu::registerConvertTritonCPUToLLVM();
+  mlir::triton::cpu::registerAllocateSharedMemoryCPU();
+  mlir::triton::cpu::registerConvertMaskedOpsToLLVM();
+  mlir::triton::cpu::registerMakePersistentKernelPass();
+
+  // Tenstorrent
+  mlir::triton::npu::registerTritonTenstorrentConvertTensorDescToLoadStore();
+  mlir::triton::npu::registerTritonTenstorrentAccelerateMatmul();
+  mlir::triton::npu::registerTritonTenstorrentPropagateTileEncoding();
+  mlir::triton::npu::registerTritonTenstorrentConvertComputeOps();
+  mlir::triton::npu::registerConvertTritonNPUToTTKernel();
+  mlir::triton::npu::registerFinalizeCBTransaction();
+  mlir::triton::npu::registerTritonTenstorrentRemoveRedundantMasks();
+  mlir::triton::npu::registerTritonTenstorrentCanonicalizeMatmulLoops();
+  mlir::triton::cpu::registerCoreSpecialize();
+
   registry.insert<
       mlir::triton::TritonDialect, mlir::cf::ControlFlowDialect,
       mlir::triton::nvidia_gpu::TritonNvidiaGPUDialect,
@@ -147,5 +174,9 @@ inline void registerTritonDialects(mlir::DialectRegistry &registry) {
       mlir::triton::amdgpu::TritonAMDGPUDialect,
       mlir::triton::proton::ProtonDialect,
       mlir::triton::proton::gpu::ProtonGPUDialect, mlir::ROCDL::ROCDLDialect,
-      mlir::triton::gluon::GluonDialect>();
+      mlir::triton::gluon::GluonDialect, mlir::triton::cpu::TritonCPUDialect,
+      mlir::triton::npu::tt::TritonTenstorrentDialect,
+      mlir::tt::ttkernel::TTKernelDialect,
+      mlir::tt::ttcore::TTCoreDialect,
+      mlir::func::FuncDialect>();
 }
